pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -307,67 +318,75 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
+				replset.LivenessProbe.Probe.Exec.Command[0] = "/data/db/mongodb-healthcheck"
 				if cr.CompareVersion("1.7.0") >= 0 {
-					replset.LivenessProbe.Probe.Handler.Exec.Command =
-						append(replset.LivenessProbe.Probe.Handler.Exec.Command,
+					replset.LivenessProbe.Probe.Exec.Command =
+						append(replset.LivenessProbe.Probe.Exec.Command,
 							"--ssl", "--sslInsecure",
 							"--sslCAFile", "/etc/mongodb-ssl/ca.crt",
 							"--sslPEMKeyFile", "/tmp/tls.pem")
 				}
 			}
+
+			if cr.CompareVersion("1.4.0") >= 0 && !replset.LivenessProbe.CommandHas(startupDelaySecondsFlag) {
+				replset.LivenessProbe.Exec.Command = append(
+					replset.LivenessProbe.Exec.Command,
+					startupDelaySecondsFlag, strconv.Itoa(replset.LivenessProbe.StartupDelaySeconds))
+			}
 		}
 
-		if gte140 && !replset.LivenessProbe.CommandHas(startupDelaySecondsFlag) {
-			replset.LivenessProbe.Handler.Exec.Command = append(
-				replset.LivenessProbe.Handler.Exec.Command,
-				startupDelaySecondsFlag, strconv.Itoa(replset.LivenessProbe.StartupDelaySeconds))
+		if replset.LivenessProbe.InitialDelaySeconds == 0 {
+			replset.LivenessProbe.InitialDelaySeconds = initialDelaySecondsDefault
+		}
+		if replset.LivenessProbe.TimeoutSeconds == 0 {
+			replset.LivenessProbe.TimeoutSeconds = timeoutSecondsDefault
+		}
+		if replset.LivenessProbe.PeriodSeconds == 0 {
+			replset.LivenessProbe.PeriodSeconds = periodSecondsDefault
+		}
+		if replset.LivenessProbe.SuccessThreshold == 0 {
+			replset.LivenessProbe.SuccessThreshold = successThresholdDefault
+		}
+		if replset.LivenessProbe.FailureThreshold == 0 {
+			replset.LivenessProbe.FailureThreshold = failureThresholdDefault
 		}
 
 		if replset.ReadinessProbe == nil {
-			replset.ReadinessProbe = &corev1.Probe{
-				Handler: corev1.Handler{
-					TCPSocket: &corev1.TCPSocketAction{
-						Port: intstr.FromInt(int(cr.Spec.Mongod.Net.Port)),
-					},
-				},
+			replset.ReadinessProbe = &corev1.Probe{}
+		}
+
+		if replset.ReadinessProbe.TCPSocket == nil {
+			replset.ReadinessProbe.TCPSocket = &corev1.TCPSocketAction{
+				Port: intstr.FromInt(int(cr.Spec.Mongod.Net.Port)),
 			}
 		}
+
 		if replset.ReadinessProbe.InitialDelaySeconds == 0 {
-			replset.ReadinessProbe.InitialDelaySeconds = int32(10)
+			replset.ReadinessProbe.InitialDelaySeconds = 10
 		}
 		if replset.ReadinessProbe.TimeoutSeconds == 0 {
-			replset.ReadinessProbe.TimeoutSeconds = int32(2)
+			replset.ReadinessProbe.TimeoutSeconds = 2
 		}
 		if replset.ReadinessProbe.PeriodSeconds == 0 {
-			replset.ReadinessProbe.PeriodSeconds = int32(3)
+			replset.ReadinessProbe.PeriodSeconds = 3
+		}
+		if replset.ReadinessProbe.SuccessThreshold == 0 {
+			replset.ReadinessProbe.SuccessThreshold = 1
 		}
 		if replset.ReadinessProbe.FailureThreshold == 0 {
-			replset.ReadinessProbe.FailureThreshold = int32(8)
+			replset.ReadinessProbe.FailureThreshold = 8
+			if cr.CompareVersion("1.11.0") >= 0 && replset.Name == ConfigReplSetName {
```

Comment:
Can I set custom values  for readiness/liveness probes   separately for all these actions ?
spec.replsets.[]
spec.sharding.mongos
spec.sharding.configsvrReplSet

Reply:
yes. added default values to cr.yaml in d14d72e.
but `LivenessProbe.SuccessThreshold` are ignored and always `1` as described in [doc](https://pkg.go.dev/k8s.io/api@v0.18.0/core/v1#Probe)